pkg/receive/hashring.go

```diff
@@ -110,9 +110,14 @@ type ketamaHashring struct {
 	numEndpoints uint64
 }
 
-func newKetamaHashring(endpoints []string, sectionsPerNode int, replicationFactor uint64) *ketamaHashring {
+func newKetamaHashring(endpoints []string, sectionsPerNode int, replicationFactor uint64) (*ketamaHashring, error) {
 	numSections := len(endpoints) * sectionsPerNode
 
+	if len(endpoints) < int(replicationFactor) {
+		return nil, fmt.Errorf("ketama: amount of endpoints needs to be larger than replication factor")
```

Comment:
We prefer using `errors.New` since it includes a stacktrace when printed out.

Reply:
Good point, that was just out of habit; ill convert it after work!

Reply:
changed it to `errors.New`